src/backend/marsha/core/tests/test_lti.py

```diff
@@ -29,6 +31,60 @@ def setUp(self):
         super().setUp()
         self.factory = RequestFactory()
 
+    def test_lti_request(self):
+        """Simulate an LTI launch request with oauth in the body.
+
+        This test uses the oauthlib library to simulate an LTI launch request and make sure
+        that our LTI verification works.
+        """
+        consumer_site = ConsumerSiteLTIPassportFactory(consumer_site__name="testserver")
+        lti_parameters = {
+            "resource_link_id": "df7",
+            "context_id": "course-v1:ufr+mathematics+0001",
+            "roles": "Instructor",
+        }
+        url = "http://testserver/lti-video/"
+        client = oauth1.Client(
+            client_key=consumer_site.oauth_consumer_key,
+            client_secret=consumer_site.shared_secret,
+        )
+        # Compute Authorization header which looks like:
+        # Authorization: OAuth oauth_nonce="80966668944732164491378916897",
+        # oauth_timestamp="1378916897", oauth_version="1.0", oauth_signature_method="HMAC-SHA1",
+        # oauth_consumer_key="", oauth_signature="frVp4JuvT1mVXlxktiAUjQ7%2F1cw%3D"
+        _uri, headers, _body = client.sign(
+            url,
+            http_method="POST",
+            body=lti_parameters,
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+        )
+
+        # Parse headers to pass to template as part of context:
+        oauth_dict = {
+            k: v
+            for k, v in [
+                param.strip().replace('"', "").split("=")
+                for param in headers["Authorization"].split(",")
+            ]
+        }
+        signature = oauth_dict["oauth_signature"]
+        oauth_dict["oauth_signature"] = unquote(signature)
+        oauth_dict["oauth_nonce"] = oauth_dict.pop("OAuth oauth_nonce")
+
+        lti_parameters.update(oauth_dict)
+        request = self.factory.post(url, lti_parameters)
+        lti = LTI(request)
+        self.assertTrue(lti.verify())
```

Comment:
For someone like me that doesn't know python at all, this assert is really disturbing. `lti.verify` return a string or raises an exception otherwise. So why `assertTrue` is working ? :-)
After reading the documentation (https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue) I understand that this assert test the value after casting the value in boolean. So it is doesn't test that the value is `true` but that the value is truthy which is not the same think from my point of view.

For this particular test it can be ok, you assert that `verify` doesn't raises an exception, but nowhere is this class the real return of verify seems really check.

Reply:
well spotted! Actually this is a remain of the time when verifiy was returning a boolean... It should be:
```suggestion
        self.assertEqual(lti.verify(), consumer_site)
```